aws-android-sdk-iot/src/main/java/com/amazonaws/mobileconnectors/iot/AWSIotMqttManager.java

```diff
@@ -735,6 +741,27 @@ public void connectUsingALPN(KeyStore keyStore,
         connect(keyStore, 443, statusCallback);
     }
 
+    /**
+     * Initializes the MQTT session and connects to the specified MQTT server
+     * using certificate and private key in keystore on port 8883 via the proxy specified by a
+     * host and port combination. Keystore should be created using IotKeystoreHelper to setup the
+     * certificate and key aliases as expected by the underlying socket helper library.
+     *
+     * @param keyStore A keystore containing an keystore with a certificate and
+     *            private key. Use IotKeystoreHelper to get keystore.
+     * @param host hostname of the proxy
+     * @param port proxy port number
+     * @param statusCallback When new MQTT session status is received the
+     *            function of callback will be called with new connection
+     *            status.
+     */
+    public void connectWithProxy(KeyStore keyStore, final String host, final int port,
+                                 final AWSIotMqttClientStatusCallback statusCallback) {
+        this.proxyHost = host;
+        this.proxyPort = port;
+        connect(keyStore, 8883, statusCallback);
```

Comment:
Is there any request from customer to connect to 443 or to any other port that IoT supports with proxy? My main concern is should the port 8883 be configurable too?

Reply:
> Is there any request from customer to connect to 443 or to any other port that IoT supports with proxy? My main concern is should the port 8883 be configurable too?

I wondered that too, but my assumption was that the proxy would have permissions to connect out on non-443 ports. If we support other ports, we'll need to add ALPN negotiation to the proxy layer as well as arbitrarily configurable ports to the underlying SDK.